### ISSUE TITLE
Dynamic texture groups fail to load (returns -1) at game start

### ISSUE THREAD

**Issue by patchuby** [YoYoGames/GameMaker-Bugs#1795]
### Description
Loading dynamic texture group near the game start (a few frames after) fails.
Loading the same texture group later doesn't seem to cause issues (when changing graphics fidelity and reloading groups).

### Steps To Reproduce
Project download link : [To come, provided via helpdesk]

To test that the loading fails at game start :
- Place the included files in Appdata>Local>Chronique_des_Silencieux
- Launch the game, I've put an error message that should trigger from the start showing the group that failed to load

To test that the loading works after game start:
- Comment out the line 91 in the script "Settings" to disable the error :
- Launching the game this way will show all sprites that are dynamic not being drawn (at the moment only backgrounds)
- Open the settings menu with escape or by clicking the watch on the top right, then change the graphics level. This calls the "settings_apply_graphicsLevel" function again, this time it should show the backgrounds, the texture groups have loaded correctly.

The starting graphics level (and so texture groups) doesn't change anything, it will always fail on start and show up later AFAIK.

[7b8f465b-f178-4fac-80f2-fabbb3ac4656](https://api.gamemaker.io/api/github/downloads/7b8f465b-f178-4fac-80f2-fabbb3ac4656)

**Comment by patchuby** [comment]
Correction : the textures actually don't fail to load when the function returns -1. So the bug is with the function rather than with the dynamic texture system. I can't upload a new project right now, and since it's the fourth time I've reported this bug, it can wait I guess.

**Comment by YYDan** [comment]
You might be able to recreate this issue using the recent copy of that game which is linked to https://github.com/YoYoGames/GameMaker-Bugs/issues/1774

**Comment by yymrennie** [comment]
We previously investigated this in relation to helpdesk ticket 211310 and found that it was failing because it was trying to load a texture group which wasn't set as Dynamic (and hence texturegroup_load() was returning -1 as no load was initiated). However the return value of texturegroup_load() has been modified to return success in this situation (since the texture group is actually loaded) but also output a warning message to indicate that what is trying to be done isn't really valid. This change will be in an upcoming beta version.

With the version of the project associated with that helpdesk ticket the game's loading check then fails later as texturegroup_load() is called with a string which doesn't match any texture group name, which is a valid failure.

As I don't believe we have the version of the project which matches this specific issue (1795) we can't confirm whether this version matches the behaviour of the previous version - if the issue is different we will probably need a matching version of the project to investigate further.

**Comment by patchuby** [comment]
You're correct, the project linked with #1774 won't be right for confirming this issue as I commented out the error message code.

Since then I modified my code for the game to wait for the dynamic texture group to load before proceeding further. I've also fixed the function being called on a  non-existant texture group (it was unfortunately the first one to be called so QA understandably thought this was the issue). 

However it will still always first return -1 on texturegroup_load() at first. I now know I can safely ignore this and keep trying to load until it works.

Best I can do from here is confirm the issue is real with correctly configured texture groups where I am with these screenshots :

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/9586594/097e13a6-cd5b-43a5-9b00-e44c4921a769)

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/9586594/7a37ff37-56a9-4bb9-9995-4e6ee86625a8)

If I comment out the show_error line, the game and the dynamic textures will load just fine.
